src/ruby/puppetserver-lib/puppet/server/http_client.rb

```diff
@@ -44,17 +44,18 @@ def self.settings
     @settings ||= {}
   end
 
-  def initialize(server, port, options = {})
+  def initialize(options = {})
     options = OPTION_DEFAULTS.merge(options)
+  end
 
-    @server = server
-    @port = port
-    @use_ssl = options[:use_ssl]
-    @protocol = @use_ssl ? "https" : "http"
+  def get(url, headers: {}, params: {}, options: {}, &block)
+    request_options = create_common_request_options(url, headers, params, options)
+    response = self.class.client_get(request_options)
+    Puppet::Server::HttpResponse.new(response, url)
```

Comment:
Looks like the return here doesn't match the API, it needs to either return just the body or yield the response object.

Reply:
I think the client docs are incorrect (and based on an earlier code). The client's `get`, etc methods should return the response when a block is not provided https://github.com/puppetlabs/puppet/blob/c3a37a92d5d08d097160a8bebc8b96d9c1b89402/spec/unit/http/client_spec.rb#L156 or if it is provided https://github.com/puppetlabs/puppet/blob/c3a37a92d5d08d097160a8bebc8b96d9c1b89402/lib/puppet/http/service/file_server.rb#L148.

Reply:
Ah okay, I missed what `execute_streaming` actually does. Cool that makes it easier.